docs/examples/basic-price-oracle/main.go

```diff
@@ -101,6 +101,69 @@ func isSupportedSubjectAsset(subjectAsset *oraclerpc.AssetSpecifier) bool {
 	return false
 }
 
+// getPurchaseRate returns the buy (purchase) rate for the asset. The unit of
+// the rate is the number of TAP asset units per BTC.
+//
+// Suppose our TAP asset is a USD stablecoin. To support liquidity and precision
+// in rate conversion, we mint 1,000,000 TAP asset units per 1.00 USD. Wallet
+// software must therefore display 1 USD for every 1,000,000 TAP asset units.
+// In effect, wallet software must divide TAP asset units by 1,000,000 (= 10^6)
+// to show the correct USD value. We call the exponent in this conversion factor
+// (in this example it's 6) the `decimalDisplay` of the asset. The
+// decimal display for a given asset is immutable and defined at minting.
+//
+// All rates returned by the price oracle service to tapd nodes are expressed as
+// TAP asset units per BTC.
+//
+// Suppose the real-world price of 1 BTC is $42,000.16. To express this as a
+// rate in TAP asset units per BTC, multiply by the decimal display conversion
+// factor (10^decimalDisplay):
+//
+// realWorldPrice      = 42,000.16
+// decimalDisplay      = 6
+// tapAssetUnitsPerBtc = realWorldPrice * (10^decimalDisplay)
+//
+//	= 42,000.16 * 1,000,000
+//	= 42,000,160,000
+//
+// Therefore, the price oracle should return the rate:
+//
+// rfqmath.NewBigIntFixedPoint(42_000_160_000, 0)
+//
+// ## When is the FixedPoint representation useful?
+//
+// Suppose another TAP asset is highly valuable, and the buy rate is
+// 0.00001 TAP asset units per BTC. Spending 1 BTC would return 0.00001 TAP
+// asset units of our valuable asset.
+//
+// To maintain integer-based communication, representing such small fractional
+// rates directly is impractical due to floating-point precision issues.
+// Instead, the price oracle returns a FixedPoint representation with integer
+// components.
+//
+// A FixedPoint number `F` consists of a coefficient `C` and scale exponent `s`:
+//
+// F = C * (10^-s)
```

Comment:
This confused me for a bit, since the actual calculation we do is: $$F = \frac {C} {10^s}$$
Which is of course equivalent to $$F = C * 10^{-s}$$
Perhaps we can write both here?

Reply:
I agree, it also took me a while. Writing both equations definitely helps make it clearer.